src/nginx/config.cc

```diff
@@ -593,7 +602,15 @@ ngx_int_t ngx_esp_build_server_config(ngx_conf_t *cf, ngx_esp_loc_conf_t *lc,
 
   // validate rewrite rule syntax
   if (config.has_api_service_config()) {
-    std::unique_ptr<std::regex> rule_check;
+    pcre *regex_compiled_;
```

Comment:
this function is getting too big,  please create a new function to handle rewrite()